### PR TITLE
Do Not Merge: generate toolchain with NoncopyableGenerics

### PR DESCRIPTION
**Pull request by kavon**

### CODE REVIEW

**Comment by kavon** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by kavon** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by kavon** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by kavon** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by kavon** [comment]
```
[8/18] Compiling MacroExamplesImplementation DictionaryIndirectionMacro.swift
/Users/ec2-user/jenkins/workspace/swift-PR-toolchain-macos/branch-main/swift-syntax/Examples/Sources/MacroExamples/Implementation/ComplexMacros/OptionSetMacro.swift:182:56: warning: 'flatMap' is deprecated: Please use compactMap(_:) for the case where closure returns an optional value
    let caseElements = optionsEnum.memberBlock.members.flatMap { member in
                                                       ^
/Users/ec2-user/jenkins/workspace/swift-PR-toolchain-macos/branch-main/swift-syntax/Examples/Sources/MacroExamples/Implementation/ComplexMacros/OptionSetMacro.swift:182:56: note: use 'compactMap(_:)' instead
    let caseElements = optionsEnum.memberBlock.members.flatMap { member in
                                                       ^~~~~~~
                                                       compactMap
/Users/ec2-user/jenkins/workspace/swift-PR-toolchain-macos/branch-main/swift-syntax/Examples/Sources/MacroExamples/Implementation/ComplexMacros/OptionSetMacro.swift:195:38: error: value of type '[EnumCaseElementSyntax]' has no member 'name'
      \(access) static let \(element.name): Self =
                             ~~~~~~~ ^~~~
Error: Command failed with non-zero exit code 1:
```

**Comment by kavon** [comment]
https://github.com/apple/swift-syntax/pull/2490
@swift-ci Please Build Toolchain macOS Platform

**Comment by kavon** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by kavon** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by kavon** [comment]
Also an issue building `swift-markdown`:

```
/Users/ec2-user/jenkins/workspace/swift-PR-toolchain-macos/branch-main/swift-markdown/Sources/Markdown/Base/RawMarkup.swift:95:24: error: type 'RawMarkup' does not conform to protocol 'Copyable'
final class RawMarkup: ManagedBuffer<RawMarkupHeader, RawMarkup> {
                       ^
```

**Comment by kavon** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by kavon** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by kavon** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by glessard** [comment]
https://ci.swift.org/job/swift-PR-toolchain-macos/1090/artifact/branch-main/swift-PR-71397-1090-osx.tar.gz

File: stdlib/cmake/modules/SwiftSource.cmake [hunk 1]
```diff
@@ -610,6 +610,7 @@ function(_compile_swift_files
   if(SWIFT_ENABLE_EXPERIMENTAL_NONCOPYABLE_GENERICS)
     list(APPEND swift_flags "-enable-experimental-feature" "NoncopyableGenerics")
     list(APPEND swift_flags "-Xfrontend" "-enable-experimental-associated-type-inference")
+    list(APPEND swift_flags "-Xfrontend" "-disable-round-trip-debug-types") # NOTE: temporary until we fix mangling!
```
Comment (tbkka) [inline stdlib/cmake/modules/SwiftSource.cmake:613]: Should we be able to remove this from this PR now?

**Pull request closed without merging**